internal/impl/packages.go

```diff
@@ -40,8 +40,19 @@ func (d *Devbox) Add(ctx context.Context, pkgsNames ...string) error {
 
 	// Only add packages that are not already in config. If same canonical exists,
 	// replace it.
-	pkgs := []*devpkg.Package{}
-	for _, pkg := range devpkg.PackageFromStrings(lo.Uniq(pkgsNames), d.lockfile) {
+	pkgs := devpkg.PackageFromStrings(lo.Uniq(pkgsNames), d.lockfile)
+	for _, pkg := range pkgs {
+		// Resolving here ensures we allow insecure before running ensurePackagesAreInstalled
+		// which will call print-dev-env. Resolving does not save the lockfile, we
+		// save at the end when everything has succeeded.
+		p, err := d.lockfile.Resolve(pkg.Raw)
+		if err != nil {
+			return err
+		}
+		if d.allowInsecureAdds {
+			p.AllowInsecure = true
```

Comment:
So this will add allow insecure field to all the packages in the add command, even though some of them may not need allow insecure?

Reply:
Yeah. The idea is that the flag affects all packages being added in that command. e.g.

`devbox add curl go --allow-insecure` will mark both of them as allow insecure. If user only wants it to apply to one of them, they should add one at a time. 

I think this is a reasonable tradeoff, otherwise we probably end up needing a new cli command e.g. `devbox allow-insecure <pkg>` or something like that.